agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/AzureMonitorLogProcessor.java

```diff
@@ -3,22 +3,36 @@
 
 package com.microsoft.applicationinsights.agent.internal.init;
 
+import com.azure.core.util.logging.ClientLogger;
 import com.azure.monitor.opentelemetry.exporter.implementation.AiSemanticAttributes;
 import com.azure.monitor.opentelemetry.exporter.implementation.OperationNames;
+import com.azure.monitor.opentelemetry.exporter.implementation.SemanticAttributes;
 import io.opentelemetry.api.trace.Span;
 import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.instrumenter.LocalRootSpan;
+import io.opentelemetry.sdk.internal.AttributesMap;
 import io.opentelemetry.sdk.logs.LogRecordProcessor;
 import io.opentelemetry.sdk.logs.ReadWriteLogRecord;
 import io.opentelemetry.sdk.trace.ReadableSpan;
+import java.lang.reflect.Field;
+import javax.annotation.Nullable;
 
 public class AzureMonitorLogProcessor implements LogRecordProcessor {
 
+  private static final ClientLogger logger = new ClientLogger(AzureMonitorLogProcessor.class);
+
+  private static final Class<?> sdkReadWriteLogRecordClass = getSdkReadWriteLogRecordClass();
+  private static final Field lockField = getLockField();
+  private static final Field attributesMapField = getAttributesMapField();
```

Comment:
I find it a bit complicated when static field initialization order matters (e.g. in this case `getLockField()` uses `sdkReadWriteLogRecordClass`), I'd suggest:

```suggestion
  private static final Field lockField;
  private static final Field attributesMapField;

  static {
    Class<?> sdkReadWriteLogRecordClass = getSdkReadWriteLogRecordClass();
    lockField = getLockField(sdkReadWriteLogRecordClass);
    attributesMapField = getAttributesMapField(sdkReadWriteLogRecordClass);
  }
```